src/t_hash.c

```diff
@@ -193,13 +204,36 @@ int hashTypeExists(robj *o, sds field) {
  * HASH_SET_COPY corresponds to no flags passed, and means the default
  * semantics of copying the values if needed.
  *
+ * The set of input flags change the command behavior.
+ *
+ * The input flags are the following:
+ *
+ * HASH_NX: Perform the operation only if the element does not exist.
+ * HASH_XX: Perform the operation only if the element already exist.
+ * HASH_CH: Modify the return value from the number of new elements added,
+ *          to the total number of elements change.
+ *
+ * The returned flags are the following:
+ *
+ * HASH_NOP: No operation was performed because of NX or XX.
+ * HASH_ADDED: The element was added (not present before the call).
+ * HASH_UPDATED: The element score was updated.
+ *
+ * Return value: todo not sure about the return value?
  */
 #define HASH_SET_TAKE_FIELD (1<<0)
 #define HASH_SET_TAKE_VALUE (1<<1)
 #define HASH_SET_COPY 0
-int hashTypeSet(robj *o, sds field, sds value, int flags) {
+int hashTypeSet(robj *o, sds field, sds value, int flags, int in_flags, int *out_flags) {
```

Comment:
it's odd that this function has both `flags` and `in_flags`.
why do we need two sets of input flags? i suppose we should combine them into one.
probably either deleting `HASH_SET_COPY` (it's clear that 0 is the default for no flags), or renaming it to `HASH_SET_DEFAULT`?

actually, i see this is modeled after `zsetAdd`, right?
so let's continue on that path, any reason they should not be more similar (only two sets of flags, and use the out_flags instead of a CH in flag)?